engine/runtime/src/main/scala/org/enso/compiler/pass/IRPass.scala

```diff
@@ -115,52 +114,14 @@ object IRPass {
     * they are guaranteed to have `restoreFromSerialization` called after they
     * have been deserialized and before any other operations occur.
     */
-  trait Metadata extends ProcessingPass.Metadata {
+  trait IRMetadata extends ProcessingPass.Metadata {
+
+    type Metadata = IRMetadata
+    type Compiler = org.enso.compiler.Compiler
 
     /** The name of the metadata as a string. */
     val metadataName: String
 
-    /** Prepares the metadata for serialization.
-      *
-      * Metadata prepared for serialization should not contain any links that
-      * span more than one module, or any other properties that are problematic
-      * when serialized.
-      *
-      * Due to the type safety properties of
-      * [[org.enso.compiler.core.ir.MetadataStorage]], to allow this conversion
-      * to work it must be type-refined to return `typeof this`. To that end,
-      * there is no default definition for this method.
-      *
-      * @param compiler the Enso compiler
-      * @return `this`, but prepared for serialization
-      */
-    def prepareForSerialization(compiler: Compiler): Metadata
```

Comment:
By defining `type Compiler` we no longer need to override these methods. That's great.